core/controllers/question_editor.py

```diff
@@ -71,31 +73,43 @@ class QuestionSkillLinkHandler(base.BaseHandler):
     """A handler for linking and unlinking questions to or from a skill."""
 
     @acl_decorators.can_manage_question_skill_status
-    def post(self, question_id, skill_id):
+    def post(self, question_id, skill_ids):
         """Links a question to a skill."""
         if not constants.ENABLE_NEW_STRUCTURE_EDITORS:
             raise self.PageNotFoundException
 
-        skill_domain.Skill.require_valid_skill_id(skill_id)
-        skill = skill_services.get_skill_by_id(skill_id, strict=False)
-        if skill is None:
-            raise self.PageNotFoundException(
-                'The skill with the given id doesn\'t exist.')
+        try:
+            skill_ids = json.loads(skill_ids)
+        except Exception:
+            raise self.PageNotFoundException
 
-        # TODO(vinitamurthi): Replace DEFAULT_SKILL_DIFFICULTY
-        # with a value passed from the frontend.
-        question_services.create_new_question_skill_link(
-            question_id, skill_id, constants.DEFAULT_SKILL_DIFFICULTY)
+        if (not isinstance(skill_ids, list) or not all([
+                isinstance(skill_id, basestring) for skill_id in skill_ids])):
+            raise self.PageNotFoundException
+
+        for skill_id in skill_ids:
+            skill_domain.Skill.require_valid_skill_id(skill_id)
+            skill = skill_services.get_skill_by_id(
+                skill_id, strict=False)
+            if skill is None:
+                raise self.PageNotFoundException(
+                    'The skill with the given id doesn\'t exist.')
+
+            # TODO(vinitamurthi): Replace DEFAULT_SKILL_DIFFICULTY
+            # with a value passed from the frontend.
+            question_services.create_new_question_skill_link(
+                question_id, skill_id,
+                constants.DEFAULT_SKILL_DIFFICULTY)
         self.render_json(self.values)
 
     @acl_decorators.can_manage_question_skill_status
-    def delete(self, question_id, skill_id):
+    def delete(self, question_id, skill_ids):
         """Unlinks a question from a skill."""
         if not constants.ENABLE_NEW_STRUCTURE_EDITORS:
             raise self.PageNotFoundException
 
         question_services.delete_question_skill_link(
```

Comment:
If this can delete multiple question skill links, maybe rename to `delete_question_skill_links`?